core/color.cpp

```diff
@@ -435,107 +381,54 @@ Color Color::named(const String &p_name) {
 }
 
 String _to_hex(float p_val) {
-
 	int v = Math::round(p_val * 255);
 	v = CLAMP(v, 0, 255);
-	String ret;
 
+	CharType c[3] = { 0, 0, 0 };
 	for (int i = 0; i < 2; i++) {
```

Comment:
Doesn't this only access the first two elements of this 3-element array?

Reply:
The last element of the array needs to remain as `0` (i.e. `NULL`), as when converting this CharType array to a `String`, the array is iterated through, with a pointer, and the characters are added to the string, when this `0` is reached it knows it is the end of the string and to stop. Without it, the pointer would access memory outside of this array.